### DISCUSSION TITLE
Questions / Suggestions about the benchmarks using jOOQ

### DISCUSSION THREAD

**Discussion by lukaseder** [TechEmpower/FrameworkBenchmarks#6673] [General]
I have a few questions / suggestions about the benchmarks using jOOQ:
https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Java/play2-java/play2-java-jooq-hikaricp

- Why is it using a 4 year old jOOQ version? (see https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Java/play2-java/play2-java-jooq-hikaricp/build.sbt) There are frequent improvements to the jOOQ code, also in terms of performance. I'd be curious to see results for a much more recent version
- The query producing `FortuneRecord` in `Application.fortunes()` is using the API very un-fortune-ately, so to speak: https://github.com/TechEmpower/FrameworkBenchmarks/blob/547370f/frameworks/Java/play2-java/play2-java-jooq-hikaricp/app/controllers/Application.java#L55
  - It is very unusual to re-create a ` DSLContext` instance per query execution. Especially when running in a benchmark, I would expect the `DSLContext` to be injected or otherwise cached. Its underlying `Configuration` contains a reflection cache, that can't be used in this particular case when calling `fetchInto(FortuneRecord.class)` (this applies to all methods!)
  - All `FORTUNE` columns are being projected in `select()`, so why not just use `selectFrom(FORTUNE)` to get a `FortuneRecord` without the unnecessary reflection?

**Comment by NateBrady23** [comment]
@lukaseder You're welcome to update this implementation with a PR. We don't maintain the implementations, we just review that they fall within the test rules and benchmark them. For questions about specific implementations, you'd have to reach out to the person who made the commits.

**Reply by lukaseder** [reply]
Thanks for the feedback. I will (PR), and I did (reach out).

**Reply by lukaseder** [reply]
> I will (PR)

Correction: Well, I can't say I can get anything to run out of the box on my windows machine, and I'm not convinced that spending the time is worth it.
